ledger/accountdb.go

```diff
@@ -749,6 +784,82 @@ func totalAccounts(ctx context.Context, tx *sql.Tx) (total uint64, err error) {
 	return
 }
 
+// reencodeAccounts reads all the accounts in the accountbase table, decode and reencode the account data.
+// if the account data is found to have a different encoding, it would update the encoded account on disk.
+// on return, it returns the number of modified accounts as well as an error ( if we had any )
+func reencodeAccounts(ctx context.Context, tx *sql.Tx) (modifiedAccounts uint, err error) {
+	modifiedAccounts = 0
+	scannedAccounts := 0
+
+	updateStmt, err := tx.PrepareContext(ctx, "UPDATE accountbase SET data = ? WHERE address = ?")
+	if err != nil {
+		return 0, err
+	}
+
+	rows, err := tx.QueryContext(ctx, "SELECT address, data FROM accountbase")
+	if err != nil {
+		return
+	}
+	defer rows.Close()
+
+	var addr basics.Address
+	for rows.Next() {
+		// once every 1000 accounts we scan through, update the warning deadline.
+		// as long as the last "chunk" takes less than one second, we should be good to go.
+		// note that we should be quite liberal on timing here, since it might perform much slower
+		// on low-power devices.
+		if scannedAccounts%1000 == 0 {
+			// The return value from ResetTransactionWarnDeadline can be safely ignored here since it would only default to writing the warnning
+			// message, which would let us know that it failed anyway.
```

Comment:
I think that from functional perspective, it would be better to keep ResetTransactionWarnDeadline returning an error - it makes it more practical to write unit test and ensure it's functional correctness.
In this case, the failure case would already be handled via a different log entry, so I don't think it would be an issue ( i.e. we would know about it, either way )